foundations/src/settings/mod.rs

```diff
@@ -461,24 +485,117 @@ pub fn to_yaml_file(settings: &impl Settings, path: impl AsRef<Path>) -> Bootstr
     )?)
 }
 
-/// Parse settings from YAML string.
+/// Builds [SettingsDeserializationOptions] with specified options.
+/// Any left unspecified will use defaults.
+pub struct SettingsDeserializationOptionsBuilder {
+    ignored_key_callback: Option<Box<dyn FnMut(IgnoredPath)>>,
+}
+
+impl SettingsDeserializationOptionsBuilder {
+    /// Initialize the builder without any options specified
+    pub fn new() -> Self {
+        SettingsDeserializationOptionsBuilder {
+            ignored_key_callback: None,
+        }
+    }
+
+    /// Set the callback that's called when YAML keys are not used during deserialization
+    /// into settings.
+    ///
+    /// If unspecified, the default will log each key as a warning, as long as it does not
+    /// end with [YAML_KEY_ALLOW_UNUSED_SUFFIX]. In that case, nothing will be logged.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use foundations::settings::{IgnoredPath, SettingsDeserializationOptionsBuilder};
+    ///
+    /// let builder = SettingsDeserializationOptionsBuilder::new();
+    /// builder.with_ignored_key_callback(Box::new(|path| {
+    ///     // Do something
+    /// }));
+    ///
+    /// ```
+    pub fn with_ignored_key_callback(mut self, callback: Box<dyn FnMut(IgnoredPath)>) -> Self {
+        self.ignored_key_callback = Some(callback);
+        self
+    }
+
+    /// Creates the configured [SettingsDeserializationOptions]
+    ///
+    /// Any values left unspecified will use default values.
+    pub fn build(self) -> SettingsDeserializationOptions {
+        SettingsDeserializationOptions {
+            ignored_key_callback: self.ignored_key_callback.unwrap_or_else(|| {
+                Box::new(|path| {
+                    #[cfg(feature = "logging")]
+                    if let IgnoredPath::Map { parent: _, key } = &path {
+                        if !key.ends_with(YAML_KEY_ALLOW_UNUSED_SUFFIX) {
```

Comment:
This suffix solution is rather clunky, but lets us avoid false positive in a very simple manner. The main alternative I'd prefer would be to automatically detect this scenario by inspecting the YAML, but I think that would be much more work and would likely have its own bugs. This should work well enough for now.

Reply:
Sounds reasonable.